pkg/grpc/portergrpc/portergrpc.go

```diff
@@ -0,0 +1,38 @@
+package portergrpc
+
+import (
+	"context"
+
+	pGRPC "get.porter.sh/porter/gen/proto/go/porterapis/porter/v1alpha1"
+	"get.porter.sh/porter/pkg/config"
+	"get.porter.sh/porter/pkg/porter"
+	"get.porter.sh/porter/pkg/secrets"
+	secretsplugin "get.porter.sh/porter/pkg/secrets/pluginstore"
+	"get.porter.sh/porter/pkg/storage"
+	storageplugin "get.porter.sh/porter/pkg/storage/pluginstore"
+	"google.golang.org/grpc"
+)
+
+type PorterServer struct {
+	PorterConfig *config.Config
+	pGRPC.UnimplementedPorterServer
+}
+
+func NewPorterServer(cfg *config.Config) (*PorterServer, error) {
+	return &PorterServer{PorterConfig: cfg}, nil
+}
+
+func (s *PorterServer) NewConnectionInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
```

Comment:
does this function adhere to some interface? if not, and there's no other reason to need an `interface{}` return type, let's change that to a specific type, preferably a concrete one (i.e. not an `interface{}` or any specific `interface`)

Reply:
the `interface{}` is used by the GRPC middleware.